### PR TITLE
Feat: project basic api

### PR DESCRIPTION
**Pull request by ZHallen122**
Ticket #6 
This pr include basic api for both project and project packages. **Note this didnt add authentication check !!!** Need to be done next.

### CODE REVIEW

File: backend/src/project/project-packages.resolver.ts
Comment (Sma1lboy) [inline backend/src/project/project-packages.resolver.ts:1]: why we need project packages resolver? Can we just update the project by only updating the project packages part? using patch-like way

File: backend/src/project/project-packages.service.ts
Comment (Sma1lboy) [inline backend/src/project/project-packages.service.ts:1]: ditto

File: backend/src/project/project.resolver.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+@Resolver(() => Projects)
+export class ProjectsResolver {
+  constructor(
+    private readonly projectsService: ProjectsService
+  ) {}
+
+  // -------- All the code need to extract the user id from the token and verify is the project user's or not
+  // add @GetAuthToken() token: string after test
+  @Query(() => [Projects])
+  async getUserProjects(@Args('userId') userId: string): Promise<Projects[]> {
```
Comment (Sma1lboy) [inline backend/src/project/project.resolver.ts:23]: check the decorator or adding a middleware or adding interceptor before all query get reached

File: backend/src/project/project.service.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    const project = await this.projectsRepository.findOne({ where: { id: projectId } });
+    if (!project) {
+      throw new NotFoundException(`Project with ID ${projectId} not found.`);
+    }
+
+    try {
+      // First, delete related project packages
+      await this.projectPackagesRepository.delete({ project_id: projectId });
+    
+      const result = await this.projectsRepository.delete(projectId);
```
Comment (Sma1lboy) [inline backend/src/project/project.service.ts:72]: using soft delete rather then just delete both project&project_packages

**Review by Sma1lboy — APPROVED** [review]
Otherwise, look good to me

File: backend/src/project/project-packages.resolver.ts
Comment (ZHallen122) [reply backend/src/project/project-packages.resolver.ts:1]: I think we can do patch-like way. The need for Project packages Resolver is for "Separation of Concerns".

Comment (ZHallen122) [reply backend/src/project/project-packages.resolver.ts:1]: Resolve check commit

File: backend/src/project/project.resolver.ts (hunk 1 above)
Comment (ZHallen122) [reply backend/src/project/project.resolver.ts]: lets use a guard and a decorator

File: backend/src/project/project.service.ts (hunk 2 above)
Comment (ZHallen122) [reply backend/src/project/project.service.ts]: good idea

File: backend/src/guard/project.guard.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  export class ProjectGuard  implements CanActivate {
+    constructor(
+      private readonly projectsService: ProjectsService,
+      private readonly jwtService: JwtService,
+    ) {}
+  
+    async canActivate(context: ExecutionContext): Promise<boolean> {
+      const gqlContext = GqlExecutionContext.create(context);
+      const request = gqlContext.getContext().req;
+
```
Comment (Sma1lboy) [inline backend/src/guard/project.guard.ts:21]: I think we can extract get auth header or whatever header as a utils functions

File: backend/src/project/dto/project.input.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+// DTOs for Project APIs
+import { InputType, Field, ID } from '@nestjs/graphql';
+
+@InputType()
+export class UpsertProjectInput {
+  @Field()
+  project_name: string;
+
+  @Field()
+  path: string;
```
Comment (Sma1lboy) [inline backend/src/project/dto/project.input.ts:10]: We don't have to explore path to user, this path is absolute path of storage in backend server

**Review by Sma1lboy — COMMENTED** [review]
also, just dont use exception here, exception causing program stop since we dont have exception handler, lets do logging, and return some status code with logging info

**Pull request merged**